scripts/src/main/resources/scripts/command/eclipse

```diff
@@ -357,16 +357,36 @@ function doAddPlugin() {
     fi
   fi
 }
-function doStartEclipse() {
+
+function doLaunchEclipse() {
   doConfigureEclipse -u
   echo "launching Eclipse in workspace ${WORKSPACE} at ${WORKSPACE_PATH}"
-  if doIsWindows
+  doStartEclipse gui -showlocation "${WORKSPACE}" -vmargs "${ECLIPSE_VMARGS}"
+}
+
+# $1: start eclipse in gui or log mode
+# $@: further parameters 
+function doStartEclipse() {
+  local start_mode="${1}"
+  shift
+  local eclipse_args=("$@")
+  local configuration_path="${DEVON_IDE_HOME}/software/eclipse-plugins/configuration/" 
+  local default_arguments="${ECLIPSE} -clean -data ${WORKSPACE_PATH} -keyring ~/.eclipse/.keyring -configuration ${configuration_path}"
+  if [ "${start_mode}" = "gui" ]
   then
-    # shellcheck disable=SC2086
-    start "eclipse" /B "${ECLIPSE}" -clean -data "${WORKSPACE_PATH}" -keyring ~/.eclipse/.keyring -vm "${ECLIPSE_JAVA_HOME}/bin/javaw" -showlocation "${WORKSPACE}" -vmargs ${ECLIPSE_VMARGS}
+    if doIsWindows
+    then
+      # shellcheck disable=SC2086
+      start "eclipse" /B ${default_arguments} -vm "${ECLIPSE_JAVA_HOME}/bin/javaw" "${eclipse_args[@]}"
+    else
+      # shellcheck disable=SC2086
+      ${default_arguments} -vm "${ECLIPSE_JAVA_HOME}/bin/java" "${eclipse_args[@]}"
+    fi
+  elif [ "${start_mode}" = "log" ]
+    then
+      "${ECLIPSE}" -consoleLog -nosplash -configuration "${configuration_path}" -vm "${ECLIPSE_JAVA_HOME}/bin/java" "${eclipse_args[@]}"
   else
-    # shellcheck disable=SC2086
-    "${ECLIPSE}" -clean -data "${WORKSPACE_PATH}" -keyring ~/.eclipse/.keyring -vm "${ECLIPSE_JAVA_HOME}/bin/java" -showlocation "${DEVON_IDE_HOME/*\//}/${WORKSPACE}" -vmargs ${ECLIPSE_VMARGS} &
+    echo "Please specify how you want to start eclipse, either in gui or log mode"
```

Comment:
Please use `doFail` instead of `echo` as this is clearly an internal error.

Reply:
done